proc/process.c

```diff
@@ -361,18 +364,38 @@ static void process_illegal(unsigned int n, exc_context_t *ctx)
 }
 
 
-#ifndef NOMMU
+static void assign_tls(process_t *process, tls_t *tls, void *tbss_addr)
+{
+	if (tls->tls_base != NULL) {
+		process->tls.tls_base = tls->tls_base;
+	}
+	else if (tbss_addr != NULL) {
+		process->tls.tls_base = tbss_addr;
+	}
+	else {
+		process->tls.tls_base = NULL;
+	}
+	process->tls.tdata_sz = tls->tdata_sz;
+	process->tls.tbss_sz = tls->tbss_sz;
+	process->tls.tls_sz = tls_size(&process->tls);
+	process->tls.arm_m_tls = tls->arm_m_tls;
+}
 
 
+#ifndef NOMMU
+
 /* TODO - adding error handling and unmapping of already mapped segments */
-int process_load32(vm_map_t *map, vm_object_t *o, offs_t base, void *iehdr)
+int process_load32(process_t *process, vm_map_t *map, vm_object_t *o, offs_t base, void *iehdr)
```

Comment:
Imho it would be better to pass a pointer to tls here, instead of process. Then we can easily make tls init and assignment code common between load32 and load64